src/main/java/fea/ui/Ui.java

```diff
@@ -0,0 +1,134 @@
+    private static final String TASKS_LOADED_SUCCESSFUL = "Tasks loaded successfully!";
+    private static final String TASKS_LOAD_ERROR = "Data file could not be found.\n"
+            + "A new data file will be auto generated upon insert of a task.";
+    private static final String ADD_TASK = "Got it. I've added this task:\n";
+    private static final String REMOVE_TASK = "Noted. I've removed this task:\n";
+    private static final String TASK_MARKED = "Nice! I've marked this task as done:\n";
+    private static final String TASK_UNMARKED = "Noted. I've marked this task as not done:\n";
+    private static final String TASKS_LIST = "Here are the tasks in your list:\n";
+    private static final String NO_TASKS_MATCH = "No matching tasks found.";
+    private static final String TASKS_MATCH = "Here are the matching tasks in your list:";
+    private Logger logger = Logger.getLogger(Ui.class.getName());
+
+    /**
+     * Takes in a file path and logs an attempt to load message to the user.
+     *
+     * @param filePath The file path of the data file.
+     */
+    public void logLoading(String filePath) {
+        logger.log(Level.INFO, "Trying to load tasks from {0}...", filePath);
+    }
+
+    /**
+     * Logs a successful load message to the user if tasks is not empty.
+     *
+     * @param tasks The current task list.
+     */
+    public void logSuccessfulLoad(TaskList tasks) {
+        if (tasks.size() == 0) {
+            logger.warning(NO_TASKS_LOADED);
+        } else {
+            logger.info(TASKS_LOADED_SUCCESSFUL);
+        }
+    }
+
+    /**
+     * Logs the loading error message when the data file could not be found.
+     */
+    public void logLoadingError() {
+        logger.warning(TASKS_LOAD_ERROR);
+    }
+
+    /**
+     * Prints task when it is added or removed.
+     *
+     * @param task     The task to be printed.
+     * @param taskList The current task list.
+     * @param toAdd    True if task is to be added, false otherwise.
+     * @return String The string of the printed task.
+     */
+    public String printTask(Task task, ArrayList<Task> taskList, boolean toAdd) {
+        StringBuilder message = new StringBuilder();
+        message.append(toAdd ? ADD_TASK : REMOVE_TASK);
+        message.append(String.format("%s%n", task.toString()))
+                .append(String.format("Now you have %d %s in the list.", taskList.size(),
+                        taskList.size() == 1 ? "task" : "tasks"));
+        return message.toString();
+    }
+
+    /**
+     * Prints task when it is marked as done or not done.
+     *
+     * @param task   The task to be printed.
+     * @param toMark True if task is to be marked as done, false otherwise.
+     * @return String The string of the task to be marked.
+     */
+    public String printMarkTask(Task task, boolean toMark) {
+        StringBuilder message = new StringBuilder();
```

Comment:
Perhaps you would want to create one instance of StringBuilder for the entire class, then calling sb.setLength(0) at the start to clear the string builder, instead of creating multiple instances?